src/request/index.ts

```diff
@@ -18,13 +18,31 @@ export const request = async <Method extends keyof Requests>(
     throw new Error('A wallet method is required');
   }
 
-  return provider.request(method, params);
+  const response = await provider.request(method, params);
+
+  if (isRpcSuccessResponse<Method>(response)) {
+    return {
+      status: 'success',
+      result: response.result,
+    };
+  }
+
+  if (response.error.code === RpcErrorCode.USER_REJECTION) {
+    return {
+      status: 'cancelled',
+    };
+  }
+
+  return {
+    status: 'error',
+    error: response.error,
+  };
 };
 
-export const isRpcSuccessResponse = <Method extends keyof Requests>(
-  response: RpcResponse<Method>
+const isRpcSuccessResponse = <Method extends keyof Requests>(
+  response: RpcBase
 ): response is RpcSuccessResponse<Method> => {
-  return Object.hasOwn(response, 'result');
+  return Object.hasOwn(response, 'result') && !!(response as RpcSuccessResponse<Method>).result;
```

Comment:
wouldn't the second condition already include the first?

```suggestion
  return !!(response as RpcSuccessResponse<Method>).result;
```

Reply:
Nope, the first checks that the key is present on the payload. The second checks that it's defined. Though it should check with `!== undefined` actually 🤔 Will change it.

Reply:
Actually, it should be fine as is. Result should always be at least an empty object if it's successful.

Reply:
can it be !== undefined without the key being present? 🤔

Reply:
I guess hasOwn returns false if the key is inherited, but that doesn't seem likely here